modules/api/src/main/java/org/eclipse/xpanse/api/controllers/CspServiceDeployApi.java

```diff
@@ -0,0 +1,74 @@
+import org.eclipse.xpanse.modules.deployment.ServiceDetailsViewManager;
+import org.eclipse.xpanse.modules.models.common.enums.Category;
+import org.eclipse.xpanse.modules.models.common.enums.Csp;
+import org.eclipse.xpanse.modules.models.service.enums.ServiceDeploymentState;
+import org.eclipse.xpanse.modules.models.service.view.DeployedService;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/** REST interface methods for processing OCL. */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse")
+@CrossOrigin
+@Secured({ROLE_CSP})
+@ConditionalOnProperty(name = "enable.agent.api.only", havingValue = "false", matchIfMissing = true)
+public class CspServiceDeployApi {
+
+    @Resource private ServiceDetailsViewManager serviceDetailsViewManager;
+
+    /**
+     * List all deployed services by a user of CSP.
+     *
+     * @return list of all services deployed by a user.
+     */
+    @Tag(name = "Service", description = "APIs to manage the services")
+    @Operation(description = "List all deployed services by a user.")
+    @GetMapping(value = "/services/csp", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @AuditApiRequest(methodName = "getCspFromRequestUri")
+    public List<DeployedService> listDeployedServicesOfCsp(
+            @Parameter(name = "categoryName", description = "category of the service")
+                    @RequestParam(name = "categoryName", required = false)
+                    Category category,
+            @Parameter(name = "cspName", description = "name of the cloud service provider")
+                    @RequestParam(name = "cspName", required = false)
+                    Csp csp,
+            @Parameter(name = "serviceName", description = "name of the service")
+                    @RequestParam(name = "serviceName", required = false)
+                    String serviceName,
+            @Parameter(name = "serviceVersion", description = "version of the service")
+                    @RequestParam(name = "serviceVersion", required = false)
+                    String serviceVersion,
+            @Parameter(name = "serviceState", description = "deployment state of the service")
+                    @RequestParam(name = "serviceState", required = false)
+                    ServiceDeploymentState serviceState) {
+        return this.serviceDetailsViewManager.listDeployedServicesOfCsp(
```

Comment:
we must remove the word list from all methods.
better would be - getAllDeployedServicesByCsp